Sources/NIOCore/SocketOptionProvider.swift

```diff
@@ -140,6 +142,7 @@ public protocol SocketOptionProvider: _NIOPreconcurrencySendable {
 //
 // You are welcome to add more helper methods here, but each helper method you add must be tested.
 extension SocketOptionProvider {
+#if !os(WASI)
```

Comment:
How about adding a typealias like the following instead of guarding out the whole method definition?

```swift
#if canImport(WASILibc)
typealias NIOLinger = Never
#else
typealias NIOLinger = linger
#endif

Reply:
Same comment as before, unsure why we'd prefer runtime failures, but this one seemed trivial so I'd considered it addressed